build_runner/lib/src/build_script_generate/build_script_generate.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:build_runner/build_runner.dart';
+import 'package:build_config/build_config.dart';
+
+const _scriptDir = '.dart_tool/build';
+const scriptLocation = '$_scriptDir/build.dart';
+
+Future<Null> ensureBuildScript() async {
+  var scriptFile = new File(scriptLocation);
+  // TODO - how can we invalidate this?
+  if (scriptFile.existsSync()) return;
+  var scriptDir = new Directory(_scriptDir);
+  if (!scriptDir.existsSync()) scriptDir.createSync(recursive: true);
+  await scriptFile.writeAsString(await _generateBuildScript());
+}
+
+Future<String> _generateBuildScript() async {
+  var result = new StringBuffer();
+  var packageGraph = new PackageGraph.forThisPackage();
+  var buildConfigs =
+      await Future.wait(packageGraph.orderedPackages.map(_packageBuildConfig));
+  result.writeln('void main() {');
```

Comment:
Should we use code_builder? If so it might make sense to start that sooner than later.

Reply:
I'm not sure - we should definitely investigate it.